packages/babel-plugin-transform-nullish-coalescing-operator/README.md

```diff
@@ -0,0 +1,50 @@
+# babel-plugin-transform-nullish-coalescing-operator
+
+> Replace `??` with an inline helper.
+
+## Example
+
+**In**
+
+```javascript
+var foo = object.foo ?? "default";
+```
+
+**Out**
+
+```javascript
+var _ref;
+var foo = (_ref = object.foo, _ref != null ? _ref : "default");
```

Comment:
Depending on how concerned we are about being *exactly* right... this isn't. `document.all ?? "default"` should give you `document.all`, not `"default"`, even though it is `== null`.

(Not familiar with the HTML Document Dot All object? `document.all` is a value which is `== null` and `== void 0` but not `===` either; it has `typeof document.all === 'undefined'` but can be called without throwing. Why? Historical reasons. Welcome to the web platform, where everything is terrible.)

If we care about this, it should instead be `ref !== null && ref !== void 0` (or however babel outputs `undefined`). Alternatively, we could chose to not care. No modern code has any legitimate reason to interact with `document.all`.

Reply:
I think we should maybe have two options here; the default, `spec`, should indeed do `ref === null && ref === undefined`, but `loose` could do `== null` if there's a performance impact.

Reply:
It is plausible that the former is faster, actually, depending on the engine. `x != null` has to end up the same as `x !== null && x !== undefined && x !== document.all`, which is more expensive than `x !== null && x !== undefined`.

Reply:
Can we just forget that `document.all` exists? 😉

Reply:
There was actual confusion at the TC39 meeting about whether `document.all` is included, so it'd be nice if Babel got this right, at least in spec mode. It would be good to get strong user feedback here.

Reply:
Get it right, as in pretend it doesn't exist? That's my preferred solution. Do we have any other syntax that explicitly accounts for `document.all`?

Reply:
`_ref !== null && _ref !== undefined` would account for it properly.

Reply:
Changed to `ref !== null && _ref !== void 0` for compliance.